spec/requests/books_spec.rb

```diff
@@ -0,0 +1,100 @@
+require 'rails_helper'
+
+RSpec.describe BooksController, type: :request do
+
+  let(:valid_params) {
+    FactoryBot.attributes_for(:book)
+  }
+
+  let(:invalid_params) {
+    { title: " " }
+  }
+
+  let(:new_params) {
+    FactoryBot.attributes_for(:book)
+  }
+
+  describe "GET #index" do
+    it "renders a successful response" do
+      Book.create! valid_params
+      
+      get books_url
+      expect(response).to be_successful
+    end
+  end
+
+  describe "GET #show" do
+    it "renders a successful response" do
+      book = Book.create! valid_params
+
+      get book_url(book)
+      expect(response).to be_successful
+    end
+  end
+
+  describe "GET #new" do
+    it "renders a successful response" do
+      get new_book_url
+      expect(response).to be_successful
+    end
+  end
+
+  describe "GET #edit" do
+    it "renders a successful response" do
+      book = Book.create! valid_params
+
+      get edit_book_url(book)
+      expect(response).to be_successful
+    end
+  end
+
+  describe "POST #create" do
+    context "with valid parameters" do
+      it "creates a new Book" do
+        expect {
+          post books_url, params: { book: valid_params }
+        }.to change(Book, :count).by(1)
+      end
+    end
+
+    context "with invalid parameters" do
+      it "does not create a new Book" do
+        expect {
+          post books_url, params: { book: invalid_params }
+        }.to change(Book, :count).by(0)
+      end
+    end
+  end
+
+  describe "PATCH #update" do
+    context "with valid parameters" do
+      it "updates the requested book" do
+        book = Book.create! valid_params
+
+        patch book_url(book), params: { book: new_params }
```

Comment:
where is a check that something was updated?

Reply:
[Done](https://github.com/kisiohlova/onlibrary/pull/8/commits/12aed2b5094390e8d5aeea6c49916958293c9afb).